org.scala-ide.sdt.core/src/org/scalaide/core/internal/compiler/ScalaPresentationCompiler.scala

```diff
@@ -49,18 +49,10 @@ import org.scalaide.util.eclipse.RegionUtils.RichRegion
 import scalariform.lexer.ScalaLexer
 import scalariform.lexer.ScalaLexerException
 import org.scalaide.core.completion.ProposalRelevanceCalculator
+import java.util.concurrent.atomic.AtomicLong
 
-class ScalaPresentationCompiler(private[compiler] val name: String, _settings: Settings) extends {
-  /*
-   * Lock object for protecting compiler names. Names are cached in a global `Array[Char]`
-   * and concurrent access may lead to overwritten names.
-   *
-   * @note This field is EARLY because `newTermName` is hit during construction of the superclass `Global`,
-   *       and the lock object has to be constructed already.
-   */
-  private val nameLock = new Object
```

Comment:
access to `nameLock` was removed in https://github.com/scala-ide/scala-ide/commit/412bee10a802e86ccb9e268fa6481615ce972a8a#diff-4279ae0b0400029aa5825551a7394a5aL308 I guess we can safely remove it.